src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -414,12 +806,12 @@ public void Dispose()
         }
 
         #endregion Private Types
-    }
 
-    internal static class EventArgsCache
-    {
-        internal static readonly PropertyChangedEventArgs CountPropertyChanged = new PropertyChangedEventArgs("Count");
-        internal static readonly PropertyChangedEventArgs IndexerPropertyChanged = new PropertyChangedEventArgs("Item[]");
-        internal static readonly NotifyCollectionChangedEventArgs ResetCollectionChanged = new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset);
+        private static class EventArgsCache
```

Comment:
This change should be reverted. This should not be nested within a generic class as doing so means there will be new instances allocated for each `T`. We only want 3 instances for all `T`'s, not 3 * N where N is the number of `T`'s.

Reply:
@justinvp 
This change wasn't introduced by me. [It's already there](https://github.com/dotnet/corefx/blob/master/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs#L419).
Should I change that in my PR?

Reply:
Your PR makes the class a subclass of `ObservableCollection<T>` (and changes `internal` into `private`) - can you revert that?

Reply:
@justinvp @karelz 
Good point, [reverted](https://github.com/dotnet/corefx/pull/26482/commits/6d0054da48eea63f22aeb3ca52d0d16fc723e129).